contributors/design-proposals/pod-priority-api.md

```diff
@@ -0,0 +1,237 @@
+* Components that are in the user-facing request serving path and must be able 
+  to scale up arbitrarily in response to load spikes (web servers, middleware,
+  etc.)
+* Important interruptible workloads that need strong guarantee of
+  schedulability and of not being interrupted
+* Less important interruptible workloads that need a less strong guarantee of
+  schedulability and of not being interrupted
+* Best effort / opportunistic
+
+### Resolving priority class names 
+
+User requests sent to Kubernetes may have "PriorityClassName" in their PodSpec.
+Admission controller resolves a PriorityCalssName to its corresponding number
+and populates the "Priority" field of the pod spec. The rest of Kubernetes
+components look at the "Priority" field of pod status and work with the integer
+value. In other words, "PriorityClassName" will be ignored by the rest of the
+system.
+
+We are going to add a new API object called PriorityClass. The priority class
+defines the mapping between the priority name and its value. It can have an
+optional description. The description is an arbitrary string and is provided
+only as a guideline for users.
+
+If a pod does not specify any "PriorityClassName", the system resolves it to 
+zero in the first version, but we will add support to annotate one of the
+`PriorityClass` objects that tells the system that it is the default one. When
+there is a default priority class, a pod with no "PriorityClassName" will get
+the value of this default class (instead of zero).
+
+```
+type PriorityClass struct {
+  metav1.TypeMeta
+  // +optional
+  metav1.ObjectMeta
+  
+  // The value of this priority class. This is the actual priority that pods
+  // recieve when they have the above name in their pod spec.
+  Value        int32
+  
+  // A description of this priority class, such as the recommended use cases.
+  // +optional
+  Description  string
+}
+```
+
+And the annotation for the default class will be something like:
+
+```yaml
+...
+annotations:
+  priorityclass.alpha.kubernetes.io/is-default-class: "true"
+...
+```
+
+### Ordering of priorities 
+
+As mentioned earlier, a PriorityClassName is resolved by the admission controller to
+its integral value and Kubernetes components use the integral value. The higher
+the value, the higher the priority.
+
+### `system` PriorityClassName
+`system` is a special priority name which is reserved and cannot be changed. It
+is used for critical system pods that must never be preempted. We set default
+policies that deny creation of pods with `system` priority. Cluster admins can
+authorize users or service accounts to create pods with this priority. When
+non-authorized users set PriorityClassName to `system` in their pos spec, their
+pod creation request will be rejected. For pods created by controllers, the
+service account must be authorized by cluster admins.
+
+### Modifying priority classes 
+
+Priority classes can be added or removed, but they cannot be updated. While
```

Comment:
if i create and delete a priority class with the same name, what is the effective difference?  we do not appear to be keeping any links back to the PriorityClass by uid to let the user know?